### PR TITLE
Turn off unnecessary logs from cleanup routines

### LINKED ISSUE #1355: Turn off unnecessary logs from cleanup routines
`make clean-all` spits out following lines which is at no use and can be discarded.

```
dpkg-query: no packages found matching docker-volume-vsphere
Error response from daemon: plugin "vsphere:latest" not found
```

Identify which lines are leftout and added redirection `> /dev/null`

### LINKED ISSUE #1556: Turnoff future timestamp warning from tar (while building managed plugin)
As seen in [CI run ](https://ci.vmware.run/vmware/docker-volume-vsphere/934), this floods CI console which makes debugging harder.

```
docker export  tempContainer  | tar -x -C /tmp/plugin/rootfs 
tar: dev/null: time stamp 2017-06-25 17:52:50 is 936835.494600484 s in the future
tar: dev/pts: time stamp 2017-06-25 17:52:50 is 936835.494084129 s in the future
tar: dev/shm: time stamp 2017-06-25 17:52:50 is 936835.493741195 s in the future
tar: etc/apk/arch: time stamp 2017-06-25 17:52:49 is 936834.493182295 s in the future
tar: etc/apk/keys: time stamp 2017-06-25 17:52:50 is 936835.492582612 s in the future
tar: etc/apk/protected_paths.d: time stamp 2017-06-25 17:52:50 is 936835.492296632 s in the future
tar: etc/apk/repositories: time stamp 2017-06-25 17:52:49 is 936834.491912526 s in the future
tar: etc/conf.d: time stamp 2017-06-25 17:52:50 is 936835.491534612 s in the future
tar: etc/crontabs: time stamp 2017-06-25 17:52:50 is 936835.491060007 s in the future
tar: etc/init.d: time stamp 2017-06-25 17:52:50 is 936835.49060046 s in the future
tar: etc/localtime: time stamp 2017-06-25 17:52:50 is 936835.490323739 s in the future
tar: etc/logrotate.d: time stamp 2017-06-25 17:52:50 is 936835.489825558 s in the future
tar: etc/modprobe.d: time stamp 2017-06-25 17:52:50 is 936835.489451766 s in the future
tar: etc/modules-load.d: time stamp 2017-06-25 17:52:50 is 936835.489177586 s in the future
tar: etc/network/if-down.d: time stamp 2017-06-25 17:52:50 is 936835.48874184 s in the future
tar: etc/network/if-post-down.d: time stamp 2017-06-25 17:52:50 is 936835.488496032 s in the future
tar: etc/network/if-post-up.d: time stamp 2017-06-25 17:52:50 is 936835.488205989 s in the future
```

### PR DESCRIPTION
**Pull request by shuklanirdesh82**
fixes #1355  Turn off unnecessary logs from cleanup routines

Note: This is follow-up change to #1554 (currently in review); so sending out this one to get proper review.

This PR is turning of following messages.

```
dpkg-query: no packages found matching docker-volume-vsphere
Error response from daemon: plugin "vsphere:latest" not found
```

Testing done: locally (`make build-all`)

```
=== Cleaning work files, images and plugin(s)...
rm -rf /tmp/plugin
rm -f docker-volume-vsphere
docker plugin rm shuklanirdesh/docker-volume-vsphere:0.16-dev -f
shuklanirdesh/docker-volume-vsphere:0.16-dev
docker rm -vf tempContainer
tempContainer
docker rmi shuklanirdesh/docker-volume-vsphere:rootfs
Untagged: shuklanirdesh/docker-volume-vsphere:rootfs
Deleted: sha256:21565646afd6d1ae650faa6fcb1a00cb18c910b2541a7d31375ae29413a964bd
Deleted: sha256:534c034383c0695ed5ffc4e931faef9990e0f606bcb20abdeb774ccc8faeef9d
Deleted: sha256:338a299fcdfc17ce0702357b5807071fad799191cc6d975b866bc18bc13d1bf5
== building Docker image, unpacking to ./rootfs and creating plugin...
cp ../build/docker-volume-vsphere .
docker build -q -t shuklanirdesh/docker-volume-vsphere:rootfs .
sha256:c7b88ac8bad307adcac14b883afadf99ae07387aaef49827d14755c1eb4f1a57
docker create --name tempContainer shuklanirdesh/docker-volume-vsphere:rootfs
bd1253e37a1d5845bd95f0f4af120c794fc15425d4de58210f28e55be58fda15
mkdir -p /tmp/plugin/rootfs
docker export  tempContainer  | tar -xm -C /tmp/plugin/rootfs
cp config.json docker-volume-vsphere /tmp/plugin
-- Creating  plugin shuklanirdesh/docker-volume-vsphere:0.16-dev ...
docker plugin create shuklanirdesh/docker-volume-vsphere:0.16-dev /tmp/plugin
shuklanirdesh/docker-volume-vsphere:0.16-dev
Pushing shuklanirdesh/docker-volume-vsphere:0.16-dev to dockerhub.io...
The push refers to a repository [docker.io/shuklanirdesh/docker-volume-vsphere]
c9ef405b7d68: Pushed
0.16-dev: digest: sha256:ffd7f8605e79474b728e935eed360b013736e5484543c28122fcdc9ecb8dc5b8 size: 519
```

/CC @msterin

### CODE REVIEW

**Comment by shuklanirdesh82** [comment]
commit#1b77ef fixes #1355

File: misc/scripts/deploy-tools.sh [hunk 1]
```diff
@@ -222,7 +222,7 @@ function cleanvm {
 function cleanupVMPre {
     case $FILE_EXT in
     deb)
-        $SSH $TARGET "$DEB_QUERY $PLUGIN_NAME > /dev/null"
+        $SSH $TARGET "$DEB_QUERY $PLUGIN_NAME > /dev/null 2>&1"
```
Comment (msterin) [inline misc/scripts/deploy-tools.sh]: this just hides error messages. Why are we getting them at the fist place?

Looking in the log ,this is likely another incarnation of CI race: 
```
tar: bin: time stamp 2017-06-25 17:52:50 is 936835.387593667 s in the future
cp config.json docker-volume-vsphere /tmp/plugin
cp: cannot stat 'config.json': No such file or directory
cp: cannot stat 'docker-volume-vsphere': No such file or directory
```
Meaning the messages are in  indication of a serious issues. 

I do not think we should turn them , we should fix the race

Comment (shuklanirdesh82) [inline misc/scripts/deploy-tools.sh]: Here is my understanding:

CI master run (passed and no indication of race): https://ci.vmware.run/vmware/docker-volume-vsphere/936

`dpkg-query -l docker-volume-vsphere`  responds back with following error message if the docker-volume-vsphere (deb package) is not present

```
=> cleanvm: Cleaning up on root@192.168.31.83 : deb Thu Jul 6 20:29:03 UTC 2017

dpkg-query: no packages found matching docker-volume-vsphere
```

The above message comes always even while running locally against ubuntu vm. We haven't remove deb package hence cleanVM routine is still having this query check first. I would say it is harmless.

Comment (msterin) [inline misc/scripts/deploy-tools.sh]: That does not explain failure to stat existing files. E.g. confg.json. it also does not explain the message from tar about time in the future.  CI race does explain both

Comment (shuklanirdesh82) [inline misc/scripts/deploy-tools.sh]: > dpkg messages are OK, tar timing ones are not.

I am going to keep changes for turning off harmless "dpkg message" and will remove the proposed change for tar command that I used .. `tar -xm` .. will revert the file and #1556 keep the issue open until we fix the race condition.

**Comment by msterin** [comment]
yes, you are right. dpkg messages are harmless and should be turned off. Nothing else in this PR, thus approving

**Comment by shuklanirdesh82** [comment]
Thanks @msterin !  I have kept only cleanup routine changes and reverted changes that I proposed for `plugin/Makefile`. I will keep the #1556 for further investigation.

**Pull request merged**